### PR TITLE
Add snaptool toolchain

### PR DESCRIPTION
**Pull request by mortenmj**
A lot of this is cribbed from rules_oci, who have a setup that makes it easy to register a toolchain implementation for every supported platform. This PR doesn't attempt to do that, but I've followed their approach since it's clean and will allow us to do this in the future.

I tested the use-binaries example by manually updating the URLs and SHAs in repo.bzl with the values from the previous release. This test is already commented out in our pipeline because of this difficulty. The other tests work just as before.

There were also some lingering rules_docker references that I've removed.

### CODE REVIEW

File: snapshots/toolchain.bzl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        "SNAPTOOL_BIN": binary.path,
+    })
+    default = DefaultInfo(
+        files = depset([binary]),
+        runfiles = ctx.runfiles(files = [binary]),
+    )
+    snaptool_info = SnaptoolInfo(binary = binary)
+    toolchain_info = platform_common.ToolchainInfo(
+        snaptool_info = snaptool_info,
+        template_variables = template_variables,
```
Comment (psalaberria002) [inline snapshots/toolchain.bzl:20]: What does template_variables do?

File: repo.bzl [hunk 2]
```diff
@@ -42,74 +51,58 @@ def _detect_host_platform(ctx):
...
-    maybe(
-        http_archive,
-        name = "io_bazel_rules_docker",
-        urls = ["https://github.com/bazelbuild/rules_docker/releases/download/v0.25.0/rules_docker-v0.25.0.tar.gz"],
-        sha256 = "b1e80761a8a8243d03ebca8845e9cc1ba6c82ce7c5179ce2b295cd36f7e394bf",
-        strip_prefix = "",
+    toolchains_repo(
+        name = snaptool_toolchain_name,
+        toolchain_type = "@com_cognitedata_bazel_snapshots//snapshots:snaptool_toolchain_type",
+        toolchain = "@{name}_snaptool//:snaptool_toolchain".format(name = name),
```
[the remark this answers is not in the document]
Comment (mortenmj) [reply repo.bzl:107]: You can register multiple toolchains, and for example use platform compatibilit to select the proper one. See for example how you can register multiple versions of go with rules go, or python with rules_python.

File: snapshots/toolchain.bzl (hunk 1 above)
Comment (mortenmj) [reply snapshots/toolchain.bzl:20]: I don't think we do, but it seems conventional to include an env var that can be used to reference the binary in script templates. Generally, TemplateVariableInfo is for exactly that. From the docs:

> Encapsulates template variables, that is, variables that can be referenced by strings like $(VARIABLE) in BUILD files and expanded by ctx.expand_make_variables and implicitly in certain attributes of built-in rules.

Comment (psalaberria002) [reply snapshots/toolchain.bzl:20]: Cool. What they call `make` variables then.

**Pull request merged**